src/python/grpcio/grpc/_adapter/_low.py

```diff
@@ -195,26 +195,30 @@ def start_batch(self, ops, tag):
         translated_op = cygrpc.operation_send_initial_metadata(
             cygrpc.Metadata(
                 cygrpc.Metadatum(key, value)
-                for key, value in op.initial_metadata))
+                for key, value in op.initial_metadata),
+            op.write_flags)
```

Comment:
No need to change anything, but a different perspective: isn't the initial metadata send the most important write we do? It's required of every RPC, and nothing happens until it's written.